### ISSUE TITLE
drupal-nginx-fpm - confused about loading custom nginx configuration

### ISSUE THREAD

**Issue by JCFerrerG** [Azure/app-service-quickstart-docker-images#265]
> Please provide us with the following information:
> ---------------------------------------------------------------
**Description**

The [entrypoint.sh#L251](https://github.com/Azure/app-service-quickstart-docker-images/blob/master/drupal-nginx-fpm/0.51/entrypoint.sh#L251) provides a mechanism for overriding the nginx config.  However, this seems to require the customer to provide the entire nginx config folder, instead of allowing them to override specific files.  

Also, [the README](https://github.com/Azure/app-service-quickstart-docker-images/tree/master/drupal-nginx-fpm/0.51#how-to-update-config-files-of-nginx) mentions something about this, but these changes will be lost when deploying to Azure.

In my opinion, this is difficult to understand and use -- even when building a custom image that uses this as the base.

This is related to https://github.com/Azure/app-service-quickstart-docker-images/issues/179 and #264.

### This issue is for a: (mark with an `x`)

- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Minimal steps to reproduce
> Try to load a custom nginx.conf from docker-compose.yml
   ```yaml
version: '3'
services:
  drupal:
    image: drupal-nginx-fpm:0.51
    volumes:
       - ./nginx.conf:/home/etc/nginx/nginx.conf
   ```

### Any log messages given by the failure
Docker logs
> 2019-03-05 18:45:51,243 INFO exited: nginx (exit status 1; not expected)

Testing the nginx.conf file
```bash
$ nginx -t -c /etc/nginx/nginx.conf
nginx: [emerg] open() "/etc/nginx/mime.types" failed (2: No such file or directory) in /etc/nginx/nginx.conf:16
nginx: configuration file /etc/nginx/nginx.conf test failed
```

### Expected/desired behavior
>Expected behavior: the container runs normally with my custom nginx.conf
>Observed behavior: nginx fails to load because of the error above

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by JCFerrerG** [comment]
Overall, I think the moving of nginx config files in entrypoint.sh is unnecessary.

What's the benefit other than moving the location to /home and creating a backup of the defaults?  
How does that make it easier for the user to supply their own configs?

Check out my [0.52_fix-265 branch](https://github.com/JCFerrerG/app-service-quickstart-docker-images/tree/0.52_fix-265/drupal-nginx-fpm/0.52) for an example that I think is easier to understand.

This does not a huge difference to what is currently in place.  The benefits are...
1. it makes things a bit easier to understand, and 
2. it resolves the error in the description.  

This is not, however, a general solution to allow the user providing their own config files (or for #264).

**Comment by LeonZhang77** [comment]
hi,  JCFerrerG...

I just published appsvcorg/drupal-nginx-fpm:0.6, please have a try.

With this version, you can modify files under /home/etc/nginx and /home/etc/varnish, these files will be persisted after restart.

Just in case if you like to restore to default configuration, please check /etc/nginx-bak and /etc/varnish-bak.

Please let me know is it fit your request? I guess you have already known how to use FTP to manage files in app service? So it should be easy. 

thanks,
-Leon
